purchases/src/test/java/com/revenuecat/purchases/PurchasesTest.kt

```diff
@@ -1252,6 +1323,31 @@ class PurchasesTest {
         }
     }
 
+    @Test
+    fun `when resetting, caches are updated`() {
+        setup()
+        every {
+            mockCache.clearLatestAttributionData(appUserId)
+        } just Runs
+        every {
+            mockIdentityManager.reset()
+        } just Runs
+        purchases.reset()
+
+        verify (exactly = 1) {
+            mockCache.setPurchaserInfoCacheTimestampToNow()
+        }
+        verify (exactly = 1) {
+            mockCache.setOfferingsCacheTimestampToNow()
+        }
+        verify (exactly = 1) {
+            mockBackend.getPurchaserInfo("fakeUserID", any(), any())
+        }
+        verify (exactly = 1) {
+            mockBackend.getOfferings("fakeUserID", any(), any())
```

Comment:
it'd be great to eventually move stuff out of Purchases, 3.5k lines of code seems way too big